compiler/ast.nim

```diff
@@ -229,7 +229,7 @@ type
   TNodeKinds* = set[TNodeKind]
 
 type
-  TSymFlag* = enum    # 47 flags!
+  TSymFlag* = enum    # 48 flags!
```

Comment:
bug:
```nim
proc fn(a: int, p1, p2: proc()) {.effectsOf: [p1, p2].}
```
gives: `Error: parameter name expected`

while the following works, but probably shouldn't (better to be strict at first on syntax, even if we later allow it; the point of disallowing it is it's simpler both for user macros and compiler when the kind is fixed to an array, and possibly allows future extensions where the `{.effectsOf: s.}` can be interpreted as a symbol s to be evaluated and would resolve to an array of params
```nim
proc fn(a: int, p1, p2: proc()) {.effectsOf: p1.}
```

Reply:
good point.